WIP Add checks for transfer-encoding header, TRACE method in RequestCodeSelector

jchannon opened the pull request:
Opening this as WIP & discussion

I was trying to find the best place to add some tests which I think is here https://github.com/openrasta/openrasta-core/blob/7c453686dbc1bb02188775961b6a2c255f4d97bf/src/OpenRasta.Tests.Unit/OperationModel/CodecSelectors/RequestCodecSelector_Specification.cs#L22 ??

serialseb commented:
I'd suggest doing tests that resemble what breaks:
 - GET request when no content-length
 - GET request when content-length is 0
 - POST request with a body with parameters when there's a content length, and when there's a TE

That should cover all bases. These days i rarely test each contributor independently, but you're of course free to do that if it makes it easier for you. Note that the tests in the old unit test project haven't been touched in a long time for the most part, so they look a bit... unfashionable.

jchannon commented:
Added tests but I can't get past the last one I added.  When it uses the POST method it is not included in the results here https://github.com/jchannon/openrasta-core/blob/35df46869febb017d31f6fd24d40b6b89b4faacd/src/OpenRasta/OperationModel/CodecSelectors/RequestCodecSelector.cs#L32 because the Post method in the handler returns false in the InputMember https://github.com/jchannon/openrasta-core/blob/8dd38c93e11262e32c794e57b5146e2e98e6c41b/src/OpenRasta/OperationModel/InputMember.cs#L17 because `isOptional` is false and `binder.IsEmpty` is true but it has a `!` check so that's false too.  Bit confused how all the internals work there.

serialseb commented:
Integrated and superceeded by 6074a8d2cd12da95a7b5856e243c90a89f554aa1
